src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java

```diff
@@ -0,0 +1,257 @@
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @NotNull
+    @Override
+    @Deprecated
+    public IBlockState getStateFromMeta(final int meta)
+    {
+        EnumFacing enumfacing = EnumFacing.getFront(meta);
+
+        if (enumfacing.getAxis() == EnumFacing.Axis.Y)
+        {
+            enumfacing = EnumFacing.NORTH;
+        }
+
+        return this.getDefaultState().withProperty(FACING, enumfacing);
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @SideOnly(Side.CLIENT)
+    @Override
+    @Deprecated
+    public boolean shouldSideBeRendered(IBlockState blockState, IBlockAccess blockAccess, BlockPos pos, EnumFacing side)
+    {
+        return true;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isOpaqueCube(final IBlockState state)
+    {
+        return false;
+    }
+
+    @NotNull
+    @Override
+    protected BlockStateContainer createBlockState()
+    {
+        return new BlockStateContainer(this, FACING);
+    }
+
+    @NotNull
+    @Override
+    @SideOnly(Side.CLIENT)
+    public BlockRenderLayer getBlockLayer()
+    {
+        return BlockRenderLayer.SOLID;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isFullCube(final IBlockState state)
+    {
+        return false;
+    }
+
+    @Override
+    public boolean isPassable(final IBlockAccess worldIn, final BlockPos pos)
+    {
+        return true;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public AxisAlignedBB getBoundingBox(final IBlockState state, final IBlockAccess source, final BlockPos pos)
+    {
+
+        if (state.getValue(FACING).equals(EAST) || state.getValue(FACING).equals(WEST))
+        {
+            return new AxisAlignedBB((float) WE_START_COLLISION_X,
+                                      (float) BOTTOM_COLLISION,
+                                      (float) WE_START_COLLISION_Z,
+                                      (float) WE_END_COLLISION_X,
+                                      (float) HEIGHT_COLLISION,
+                                      (float) WE_END_COLLISION_Z);
+        }
+        else
+        {
+            return new AxisAlignedBB((float) SN_START_COLLISION_X,
+                                      (float) BOTTOM_COLLISION,
+                                      (float) SN_START_COLLISION_Z,
+                                      (float) SN_END_COLLISION_X,
+                                      (float) HEIGHT_COLLISION,
+                                      (float) SN_END_COLLISION_Z);
+        }
+    }
+
+    @SuppressWarnings("deprecation")
+    @NotNull
+    @Override
+    public IBlockState getStateForPlacement(
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Don't override a deprecated method or explicitly mark it as "@Deprecated". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)